src/sagas/options.js

```diff
@@ -147,10 +146,11 @@ function* saveRemoteOption ([ name, value ]: any): any | ResultType {
     change: name })
 }
 
-function* saveGlobalOptions ({ payload: { sshKey, showNotifications, notificationSnoozeDuration, language, refreshInterval }, meta: { transactionId } }: SaveGlobalOptionsActionType): Generator<any, any, any> {
-  const { ssh, locale, refresh } = yield all({
+function* saveGlobalOptions ({ payload: { sshKey, showNotifications, notificationSnoozeDuration, language, refreshInterval, persistLocale }, meta: { transactionId } }: SaveGlobalOptionsActionType): Generator<any, any, any> {
+  const { ssh, locale, refresh, persistLoc } = yield all({
     ssh: call(saveSSHKey, ...Object.entries({ sshKey })),
-    locale: call(saveRemoteOption, ...Object.entries({ locale: language })),
+    locale: (persistLocale || persistLocale === undefined) ? call(saveRemoteOption, ...Object.entries({ locale: language })) : {},
```

Comment:
OK. It seems we need to check also the current value in the store (via `select()`) if persist flag was not changed.